src/Microsoft.AspNetCore.SystemWebAdapters.CoreServices/SystemWebAdaptersExtensions.cs

```diff
@@ -122,6 +118,13 @@ public Action<IApplicationBuilder> Configure(Action<IApplicationBuilder> next)
             => builder =>
             {
                 builder.UseMiddleware<SetHttpContextTimestampMiddleware>();
+                builder.UseMiddleware<RegisterAdapterFeaturesMiddleware>();
+
+                if (builder.AreHttpApplicationEventsRequired())
+                {
+                    builder.UseMiddleware<HttpApplicationMiddleware>();
```

Comment:
This is being moved so that HttpApplication is available if someone wants to manually call the output caching before we get something in. Otherwise, the call to `app.UseOutputCaching()` would have to be after `app.UseSystemWebAdapters`